### DISCUSSION TITLE
Rotating the SmoothSuperellipsoid by an arbitrary angle

### DISCUSSION THREAD

**Discussion by mnabilbhuiyan** [idaholab/moose#16162] [Q&A General] [answered]
Hello,
I am looking how to rotate the "SmoothSuperellipsoid" by an arbitrary angle say 20 with respect to x or y axis in a two dimensional grain growth simulation. I need some suggestions on how I can rotate the axis of the ellipse? I am using the following block but I am confused how I can define the angle.

[ICs]
  [./c]
    type = SpecifiedSmoothSuperellipsoidIC
    variable = c
    x_positions = '750'
    y_positions = '550'
    z_positions = '0'
    as = '120.0'
    bs = '60.0'
    cs = '60.0'
    ns = '2.0'
    invalue = 1.0
    outvalue = 0.0
    int_width = 1.0
  [../]
[]
![unnamed](https://user-images.githubusercontent.com/60020086/98878646-53f0f180-2438-11eb-821d-5a7c50e7fe21.png)

Is it possible to define the angle of the semi axis with length anyhow? 
Best Regards
Nabil

**Comment by laagesen** [comment, marked as answer]
Right now the code does not have the capability to do this. It wouldn't be too hard to add if you wanted to try it. It would be a good first MOOSE contribution if you are interested in becoming a contributor. Have a look at `SmoothSuperEllipsoidBaseIC` to see how the current version works

**Reply by mnabilbhuiyan** [reply]
Thank you for the direction
